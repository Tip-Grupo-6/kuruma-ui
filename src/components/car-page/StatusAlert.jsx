import Box from "@mui/material/Box";
import {Alert, AlertTitle, Collapse} from "@mui/material";
import IconButton from "@mui/material/IconButton";
import CloseIcon from '@mui/icons-material/Close';
import React from "react";
import {makeStyles} from "@material-ui/core/styles";

const styles = makeStyles(theme => ({
    boxAlert: {
        padding: "10px 10px 0 10px",
        [theme.breakpoints.up("sm")]: {
            padding: "25px",
            position: "absolute"
        }
    },
    statusAlert: {
        width: "300px",
    },
}));

export const StatusAlert = ({status, open, onClose}) => {
    const classes = styles()

    const severityStatus = () => {
        switch (status?.status_color) {
            case "red":
                return "error";
            case "yellow":
                return "warning";
            default:
                return "success";
        }
    }

    return (
        <Box sx={{ width: '100%' }} className={classes.boxAlert}>
            <Collapse in={open} data-testid={"collapse"}>
                <Alert data-testid={"alert"} severity={severityStatus()} className={classes.statusAlert} action={
                    <IconButton
                        aria-label="close"
                        color="inherit"
                        size="small"
                        onClick={onClose}
                    >
                        <CloseIcon fontSize="inherit" />
                    </IconButton>
                }>
                    <AlertTitle>Chequeo de {status?.name}</AlertTitle>
                    Último chequeo: {status?.last_change}<br/>
                    Próximo chequeo: {status?.next_change_due}
                </Alert>
            </Collapse>
        </Box>
    )
}