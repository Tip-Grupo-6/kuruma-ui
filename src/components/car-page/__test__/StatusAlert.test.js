import React from "react";

import renderer from 'react-test-renderer';
import {render, screen} from '@testing-library/react'

import {StatusAlert} from "../StatusAlert";

it('status alert should match with snapshot', () => {
    const component = renderer.create(
        <StatusAlert status={null} onClose={() => {}} open={false}/>
    )

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
})

describe("open alert", () => {
    it.each([
        {open: false, className: "MuiCollapse-hidden"},
        {open: true, className: "MuiCollapse-entered"},
    ])("when open is '$open' should have class '$className'", ({open, className}) => {
        render(
            <StatusAlert status={null} onClose={() => {}} open={open}/>
        )
        const container = screen.getByTestId("collapse")
        expect(container).toHaveClass(className);
    })
});

describe("check icon status", () => {
    it.each([
        {status_color: 'red', iconId: "ErrorOutlineIcon"},
        {status_color: 'yellow', iconId: "ReportProblemOutlinedIcon"},
        {status_color: 'green', iconId: "SuccessOutlinedIcon"}
    ])("when status is '$status' should have icon '$iconId'", ({status_color , iconId}) => {
        render(
            <StatusAlert status={{status_color: status_color}} onClose={() => {}} open={false}/>
        )
        const container = screen.getByTestId(iconId)
        expect(container).toBeTruthy();
    })
});

it('check message info with status', () => {
    const name = 'Aceite'
    const last_change = '01/01/2023'
    const next_change_due = '01/12/2023'
    render(
        <StatusAlert status={{name, last_change, next_change_due}} onClose={() => {}} open={false}/>
    )
    const container = screen.getByTestId('alert')
    expect(screen.getByText(`Chequeo de ${name}`)).toBeInTheDocument();
    expect(container).toHaveTextContent(`Último chequeo: ${last_change}`)
    expect(container).toHaveTextContent(`Próximo chequeo: ${next_change_due}`)
})
