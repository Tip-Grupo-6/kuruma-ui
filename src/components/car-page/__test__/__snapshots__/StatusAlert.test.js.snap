// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`status alert should match with snapshot 1`] = `
<div
  className="makeStyles-boxAlert-1 MuiBox-root css-8atqhb"
>
  <div
    className="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
    style={
      {
        "minHeight": "0px",
      }
    }
  >
    <div
      className="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
    >
      <div
        className="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
      >
        <div
          className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-standardError MuiAlert-standard makeStyles-statusAlert-2 css-1hdyuqk-MuiPaper-root-MuiAlert-root"
          role="alert"
        >
          <div
            className="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
              data-testid="ErrorOutlineIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
              />
            </svg>
          </div>
          <div
            className="MuiAlert-message css-1pxa9xg-MuiAlert-message"
          >
            <div
              className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom MuiAlertTitle-root css-yslrua-MuiTypography-root-MuiAlertTitle-root"
            >
              Chequeo de 
            </div>
            Último chequeo: 
            <br />
            Próximo chequeo: 
          </div>
          <div
            className="MuiAlert-action css-ki1hdl-MuiAlert-action"
          >
            <button
              aria-label="close"
              className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeSmall css-1e0d89p-MuiButtonBase-root-MuiIconButton-root"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                data-testid="CloseIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
